code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -180,15 +180,21 @@ var/list/gaslist_cache = null
 				//to_chat(world, "post [temperature], [cached_gases["plasma"][MOLES]], [cached_gases["co2"][MOLES]])
 			*/
 	if(holder)
-		if(cached_gases["freon"])
-			if(cached_gases["freon"][MOLES] >= MOLES_PLASMA_VISIBLE)
-				if(holder.freon_gas_act())
-					cached_gases["freon"][MOLES] -= MOLES_PLASMA_VISIBLE
-
-		if(cached_gases["water_vapor"])
-			if(cached_gases["water_vapor"][MOLES] >= MOLES_PLASMA_VISIBLE)
+		var/list/freon_cache = cached_gases["freon"]

```

Comment:
what's the point of this? it's already cached as a proc var. is the list access overhead actually significant enough that another read+write here is faster?